helix-core/src/main/java/org/apache/helix/manager/zk/ZkBucketDataAccessor.java

```diff
@@ -429,4 +427,14 @@ private List<String> getPathsToDelete(String path, List<String> staleVersions) {
     staleVersions.forEach(ver -> pathsToDelete.add(path + "/" + ver));
     return pathsToDelete;
   }
+
+  private String getLastSuccessfulWriteVersion(String path) {
+    byte[] binaryVersionToRead = _zkBaseDataAccessor.get(path + "/" + LAST_SUCCESSFUL_WRITE_KEY,
+        null, AccessOption.PERSISTENT);
+    if (binaryVersionToRead == null) {
+      throw new ZkNoNodeException(
+          String.format("Last successful write ZNode does not exist for path: %s", path));
```

Comment:
If we throw exception, the gc thread will be break, right? Shall we just log it and let it continue? Or we should stop the entire Accessor...

Reply:
I believe this try block should catch it, the one we are submitting to the GC thread:
```
         try {
            _gcTaskFutureMap.remove(rootPath);
            deleteStaleVersions(rootPath);
          } catch (Exception ex) {
            LOG.error("Failed to delete the stale versions.", ex);
          }
        }, _versionTTLms, TimeUnit.MILLISECONDS));
```
My understanding is that this should prevent the gc thread from crashing and just log the exception


`compressedBucketRead` will still throw the exception though